conda_build/build.py

```diff
@@ -288,6 +290,21 @@ def write_about_json(m, config):
             value = m.get_value('about/%s' % key)
             if value:
                 d[key] = value
+        # for sake of reproducibility, record some conda info
+        conda_info = json.loads(subprocess.check_output(['conda', 'info', '--json', '-s']))
+        d['conda_version'] = conda_info['conda_version']
+        d['conda_build_version'] = conda_info['conda_build_version']
+        d['conda_env_version'] = conda_info['conda_env_version']
+        d['offline'] = conda_info['offline']
+        d['channels'] = conda_info['channels']
```

Comment:
I tested that, and it put the token in the output.  Yikes.  I have added a commit that scrubs URLs in a pretty rudimentary way.

```
re.sub('\/t\/[a-zA-Z0-9\-]*\/', '/t/<TOKEN>/', channel)
```

This is tested with:
```
def test_sanitize_channel():
    test_url = 'https://conda.anaconda.org/t/ms-534991f2-4123-473a-b512-42025291b927/somechannel'
    assert build.sanitize_channel(test_url) == 'https://conda.anaconda.org/t/<TOKEN>/somechannel'

```